### DISCUSSION TITLE
High baseline value in searchlight analysis

### DISCUSSION THREAD

**Discussion by mz973** [rsagroup/rsatoolbox#404] [Q&A]
I'm doing some searchlight analyses on beta maps using crossnobis for between-condition distance and cosine similarity to compare RDMs, and I think this combination is recommended in the RSAtoolbox documentation for a meaningful zero point. But the results have very high baseline value which cause second-level outcome to be significant throughout the whole brain. 

My data was collected over 12 runs, but because the conditions were not balanced entirely I cannot use a leave-one-run-out cross-validation. Instead each time it was a leave-two-run-out. The noise was estimated from measurement. I'm not sure if this could be the culprit. When I switched to "spearman" for RDM comparison the result looks normal.

I wonder if some of you have encountered this issue before and can point me to the right direction? Thank you all!

**Comment by JasperVanDenBosch** [comment]
Hi Mengya!
I've moved this to the Q&A discussion section as it seems to be less about the software and more a methods question (although I could be wrong). If you find the answer could you report it here for future visitors?

When you mention the high baseline value, I'm assuming you mean on the data (not the RDM)? And what sort of second level analysis are you applying? It may help to post some screenshots here.

**Reply by mz973** [reply]
Hello Jasper,
Thank you for the quick response and moving the question to a more suitable place! I can't pinpoint the exact reason but my hunch is it has something to do with how I prepared the data(?).

Yes I was talking about the correlation coefficient between data and model RDMs when I said "high baseline value'. Normally I would see on averaged maps some regions in the negative and some in the positive range (e.g. screenshot 1, which is using crossnobis+spearman), but in my case I had only positive values throughout the brain (screenshot 2, using crossnobis+cosine). In turn this makes the group level significant result just one big blob of the whole brain mask. I'm using a non-parametric cluster-based permutation procedure for the second level, although I dont think using a parametric one would change anything.

Can you think of anything that may result in this? Cheers!

![1](https://github.com/user-attachments/assets/17c394a5-1a7d-4b85-99bf-5272906c41ce)

![screenshot 2](https://github.com/user-attachments/assets/f92ac473-938a-4a7f-ae90-7842e2fc149e)

**Reply by JasperVanDenBosch** [reply]
Can I ask what your size is for the searchlight sphere? 
If I understand correctly the statistic here is the second-level stat across participants, where in each participant you are using the fit of a single model against the data RDMs? It is typical for cosine to give a higher value in general. Have you considered comparing the model against others?

**Reply by mz973** [reply]
The radius is 3 voxels.
Yes I was referring to the second-level stat across subjects, specifically in each voxel is the t-score of a group comparison against 0 (in the screenshot it is not tscore though, just averaged correlation values across people).

I can certainly have competitive models, but I also want to understand why using cosine similarity gives high values everywhere. Shouldn't it generate meaningful zero points like the documentation has explained? And on the subject of having multiple models, I have not tried it before with searchlight class - is it as simple as combining ModelSelect with evaluate_model_searchlight? 

Much appreciate your swift replies btw!

**Reply by JasperVanDenBosch** [reply]
Afaik, the meaningful zero point applies to the dissimilarity values in the RDM, not necessarily to the model comparison. 
If you were to use ModelSelect in this context, you'd be interested in the thetas (weights) on the Result objects returned, not the model fit result (as this would be the either/or fit).
To be honest, we're not currently actively maintaining the searchlight code since we are planning to do a major rework of it in the near future. One reason for this is that model comparison isn't well integrated for the searchlight right now..
Happy to help

**Reply by mz973** [reply]
Ah ok, I was misunderstanding it. Thanks for clarifying!

My experience with the toolbox has been only with ModelFixed. I will give Modelselect a try as there is certainly a use for it too. Will report back if for some reason it doesn't run and I couldn't fix it. Cheers again.

**Reply by HeikoSchuett** [reply]
Hi @mz973 ,

just as a short note: the high baseline value you observe is entirely expected for cosine similarity. This similarity measure behaves differently from correlations. Due to all true distances being positive, all true cosine similarities between RDMs are positive and typical similarities are substantially different from 0. This makes tests whether this is significantly different from 0 meaningless. If you need such a test, either compare against a null model (like a constant RDM) or use a correlation as your similarity.

best
Heiko
